shared/login/login/index.js

```diff
@@ -75,7 +80,8 @@ export default connect(
       onForgotPassphrase: () => dispatch(openAccountResetPage()),
       onLogin: (user, passphrase, store) => dispatch(relogin(user, passphrase, store)),
       onSignup: () => dispatch(routeAppend(['signup'])),
-      onSomeoneElse: () => { dispatch(login()) }
+      onSomeoneElse: () => { dispatch(login()) },
+      onSaveInKeychainChange: (user, saveInKeychain) => { if (!saveInKeychain) dispatch(clearStoredSecret(user)) }
```

Comment:
Maybe better to call this clearStoredSecret, and move the logic here to `_onSaveInKeychainChange` just because nothing else here deals with logic